src/overlays/actors/ovl_Obj_Warp2block/z_obj_warp2block.c

```diff
@@ -43,36 +42,305 @@ const ActorInit Obj_Warp2block_InitVars = {
+        } else {
+            player->stateFlags2 |= 0x800000;
+        }
+    }
+
+    return 0;
+}
+
+
+s32 func_80BA228C(ObjWarp2block *this, GlobalContext *globalCtx) {
+    if (globalCtx->msgCtx.unk_E3EE == 4) {
+        this->actionFunc_168 = func_80BA2218;
+    }
+
+    if (globalCtx->msgCtx.unk_E3EC == 0xA) {
+        if (this->unk_172 == 0xFE) {
+            this->unk_16E = 0x6E;
+        } else {
+	        this->unk_16E = this->unk_16E - 1;
+	        if (this->unk_16E == 0) {
+	            return 1;
+	        }
+	    }
+    }
+    return 0;
+}
+
+
+s32 func_80BA2304(ObjWarp2block *this, GlobalContext *globalCtx) {
+    s32 ret;
+
+    ret = this->actionFunc_168(this, globalCtx);
+    this->unk_172 = globalCtx->msgCtx.unk_E3EC;
+
+    return ret;
+}
+
+
+void ObjWarp2block_Init(Actor* thisx, GlobalContext *globalCtx2) {
+	GlobalContext* globalCtx = globalCtx2;
+    ObjWarp2block* this = THIS;
+    CollisionHeader* collisionHeader;
+
+    collisionHeader = NULL;
+    this->dyna.actor.shape.rot.z = 0;
+    this->dyna.actor.world.rot.z = this->dyna.actor.shape.rot.z;
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+
+    Actor_SetScale(&this->dyna.actor, sSpawnData[(this->dyna.actor.params >> 8) & 1].scale);
+    this->actionFunc_168 = func_80BA2218;
+    Actor_SetFocus(&this->dyna.actor, sSpawnData[(this->dyna.actor.params >> 8) & 1].focus);
+
+    if (((this->dyna.actor.params >> 0xF) & 1)) {
+        func_80BA24E8(this);
+        if (Flags_GetSwitch(globalCtx, this->dyna.actor.params & 0x3F) != 0) {
+            this->dyna.actor.draw = NULL;
+        }
+        DynaPolyActor_Init(&this->dyna, 0);
+        CollisionHeader_GetVirtual(&D_06000B30, &collisionHeader);
+        this->dyna.bgId = DynaPoly_SetBgActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, collisionHeader);
+    } else {
+        ObjWarp2block_SetInactive(this);
+    }
+
+    osSyncPrintf((const char *) "時のブロック(ワープ２) (<arg> %04xH <type> color:%d range:%d)\n",
+    	this->dyna.actor.params & 0xFFFF,
+    	this->dyna.actor.home.rot.z & 7,
+    	((s32) this->dyna.actor.params >> 0xB) & 7);
+}
+
+void ObjWarp2block_Destroy(Actor* thisx, GlobalContext *globalCtx) {
+    ObjWarp2block* this = THIS;
+    if ((this->dyna.actor.params >> 0xF) & 1) {
+        DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    }
+}
+
+void ObjWarp2block_SetInactive(ObjWarp2block *this) {
+    this->actionFunc_164 = ObjWarp2block_DoNothing;
+    this->dyna.actor.draw = NULL;
+}
+
+void ObjWarp2block_DoNothing(ObjWarp2block *this, GlobalContext *globalCtx) {}
+
+void func_80BA24E8(ObjWarp2block *this) {
+    this->actionFunc_164 = func_80BA24F8;
+}
+
+void func_80BA24F8(ObjWarp2block *this, GlobalContext *globalCtx) {
+    Actor *current;
+
+    current = globalCtx->actorCtx.actorLists[7].head;
+
+    while (current != NULL) {
+        if (current->id == 0x1D6) {
+            if (((current->params >> 0xF) & 1) == 0) {
```

Comment:
```suggestion
            if (!((current->params >> 0xF) & 1)) {
```
The three conditionals here can be combined with &&